src/extensionsIntegrated/Emmet/main.js

```diff
@@ -0,0 +1,544 @@
+/*
+ * GNU AGPL-3.0 License
+ *
+ * Copyright (c) 2021 - present core.ai . All rights reserved.
+ * Original work Copyright (c) 2024 [emmet.io](https://github.com/emmetio/brackets-emmet).
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://opensource.org/licenses/AGPL-3.0.
+ */
+
+
+define(function (require, exports, module) {
+    const AppInit = require("utils/AppInit");
+    const PreferencesManager = require("preferences/PreferencesManager");
+    const Strings = require("strings");
+    const CodeHintManager = require("editor/CodeHintManager");
+    const {
+        markupSnippets,
+        htmlTags,
+        positiveSymbols,
+        negativeSymbols
+    } = require('./emmet-snippets');
+
+
+    /**
+     * The Emmet api's
+     */
+    const EXPAND_ABBR = Phoenix.libs.Emmet.expand;
+    // const EMMET = Phoenix.libs.Emmet.module;
+
+
+    /**
+     * A list of all the markup snippets that can be expanded.
+     * For ex: 'link:css', 'iframe'
+     * They expand differently as compared to normal tags.
+     * Refer to `./emmet-snippets.js` file for more info.
+     */
+    const markupSnippetsList = Object.keys(markupSnippets);
+
+
+    // For preferences settings, to toggle this feature on/off
+    const PREFERENCES_EMMET = "emmet";
+    let enabled = true; // by default:- on
+
+    PreferencesManager.definePreference(PREFERENCES_EMMET, "boolean", enabled, {
```

Comment:
You might need to move the define preference to core as it shared between to integrated extensions. so maybe add a new file in `src/preferences/allPreferences.js` where we define all preferences in code, and expose it via a constant in `PreferencesManager` like: `PreferencesManager.ALL_PREFERENCES.EMMET`.

We could eventually use that as a global place to put all system level preference to see all that in a single place.